lib/ecto/query/planner.ex

```diff
@@ -407,6 +407,8 @@ defmodule Ecto.Query.Planner do
   defp valid_subquery_value?(args) when is_list(args), do: false
   defp valid_subquery_value?({container, _, args})
        when container in [:{}, :%{}, :&] and is_list(args), do: false
+  defp valid_subquery_value?(nil), do: true
+  defp valid_subquery_value?(arg) when is_atom(arg), do: false
```

Comment:
We need to make sure we still allow booleans too:

```suggestion
  defp valid_subquery_value?(arg) when is_atom(arg) and not is_boolean(arg), do: false
```